server.py

```diff
@@ -0,0 +1,82 @@
+#! /usr/bin/env python3
+
+# Author: Thomas Gruber
+# Version: 0.8, 11/06/18
+
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+import io
+import json
+import time
+
+
+class ControlHandler (BaseHTTPRequestHandler):
+    SERVER_PATH = '/terradrone'
+    CHAR_ENCODING = 'utf-8'
+
+    # commands should correspond to those written in Android application
+    CMD_ERROR                                                   = 0
+    CMD_TEST                                                    = 1
+    CMD_FORWARD_START                                           = 100
+    CMD_FORWARD_STOP                                            = 101
+    CMD_BACKWARD_START                                          = 102
+    CMD_BACKWARD_STOP                                           = 103
+    CMD_ROTATE_RIGHT_START                                      = 104
+    CMD_ROTATE_RIGHT_STOP                                       = 105
+    CMD_ROTATE_LEFT_START                                       = 106
+    CMD_ROTATE_LEFT_STOP                                        = 107
+    CMD_COLLECT                                                 = 200
+
+
+    def do_GET(self):
+        # deny request
+        self.send_error(418)
```

Comment:
If you want to keep style consistent these errors should be added with the others above. This does not have to be done though, only if we have time in the project.

Reply:
It's a specific HTTP error that I'm sending back, not one I arbitrarily made. Though, I could add a simple constant to keep with styling.

Reply:
Commit 3122d08b84f914cacb9d71c4dc32b8d1b5c60c8d removed all magic numbers including HTTP response codes and port. The numbers are now in a global variable with an appropiate name to increase readability.